### PR TITLE
【昇思学习营第六期·五一篇】实践项目: 基于MindSpore实现ViT病理图像识别

### PR DESCRIPTION
**Pull request by shun-cheng**
## Description of Changes

1.The number of training epochs increased by 2, and the accuracy improved to 87.84%.
2.Adjusted the file location.
3.The model introduction is more detailed.
4.Remove embedded built-in image display.
5.Simplified the output information.

## Test Environment
Hardware: 1*ascend-snt9b|ARM: 19核 180GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1

## Test Report

Test Result
Accuracy: 87.84%
Training Time: 41.90 minutes
Eval Time: 0.33 minutes
Total Script Runtime: 43.28 minutes

## Additional Notes

In this project, the PathMNIST dataset provided by MedMNIST is used based on the MindSpore framework, utilizing a pre-trained Vision Transformer (ViT) model for the identification task of colorectal tissue images.

### CODE REVIEW

**Comment by shun-cheng** [comment]
Test Environment
Hardware: 1*ascend-snt9b|ARM: 19核 180GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1

**Comment by hongziqi** [comment]
Thank you very much for your submission! 

> ## PR Type
> > /kind ascend-case
> 
> ## Description of Changes
> 1.The number of training epochs increased by 2, and the accuracy improved to 87.84%. 2.Adjusted the file location. 3.The model introduction is more detailed. 4.Remove embedded built-in image display. 5.Simplified the output information.
> 
> ## Test Environment
> Hardware: 1*ascend-snt9b|ARM: 19核 180GB Image: python3.9-ms2.5.0-cann8.0.0.beta1
> 
> ## Test Report
> Test Result Accuracy: 87.84% Training Time: 41.90 minutes Eval Time: 0.33 minutes Total Script Runtime: 43.28 minutes
> 
> ## Additional Notes
> In this project, the PathMNIST dataset provided by MedMNIST is used based on the MindSpore framework, utilizing a pre-trained Vision Transformer (ViT) model for the identification task of colorectal tissue images.

You can also add the following content
## Related Issues
#44

**Comment by shun-cheng** [comment]
> Thank you very much for your submission! Here's a suggestion: Instead of setting notebook cell output to empty, you can control [MindSpore's log](https://www.mindspore.cn/docs/zh-CN/r2.6.0rc1/api_python/env_var_list.html?highlight=glog_v#%E6%97%A5%E5%BF%97) verbosity by configuring the log level through environment variables. For optimal results, place this code after installing dependencies but before importing MindSpore:
> 
> ```python
> import os
> # Set log level: 4(EXCEPTION), 3(ERROR), 2(WARNING), 1(INFO), 0(DEBUG)
> os.environ['GLOG_v'] = '3'
> ```
> 
> This approach provides cleaner output while preserving important information in your notebook.



> Thank you very much for your submission! Here's a suggestion: Instead of setting notebook cell output to empty, you can control [MindSpore's log](https://www.mindspore.cn/docs/zh-CN/r2.6.0rc1/api_python/env_var_list.html?highlight=glog_v#%E6%97%A5%E5%BF%97) verbosity by configuring the log level through environment variables. For optimal results, place this code after installing dependencies but before importing MindSpore:
> 
> ```python
> import os
> # Set log level: 4(EXCEPTION), 3(ERROR), 2(WARNING), 1(INFO), 0(DEBUG)
> os.environ['GLOG_v'] = '3'
> ```
> 
> This approach provides cleaner output while preserving important information in your notebook.



> Thank you very much for your submission! Here's a suggestion: Instead of setting notebook cell output to empty, you can control [MindSpore's log](https://www.mindspore.cn/docs/zh-CN/r2.6.0rc1/api_python/env_var_list.html?highlight=glog_v#%E6%97%A5%E5%BF%97) verbosity by configuring the log level through environment variables. For optimal results, place this code after installing dependencies but before importing MindSpore:
> 
> ```python
> import os
> # Set log level: 4(EXCEPTION), 3(ERROR), 2(WARNING), 1(INFO), 0(DEBUG)
> os.environ['GLOG_v'] = '3'
> ```
> 
> This approach provides cleaner output while preserving important information in your notebook.

> Thank you for your suggestion. I have modified the document according to your advice. The log output level is controlled in the environment configuration section.

**Comment by shun-cheng** [comment]
Thank you for your suggestion！I have modified the document according to your advice. The log output level is controlled in the environment configuration section.
> Thank you very much for your submission! Here's a suggestion: Instead of setting notebook cell output to empty, you can control [MindSpore's log](https://www.mindspore.cn/docs/zh-CN/r2.6.0rc1/api_python/env_var_list.html?highlight=glog_v#%E6%97%A5%E5%BF%97) verbosity by configuring the log level through environment variables. For optimal results, place this code after installing dependencies but before importing MindSpore:
> 
> ```python
> import os
> # Set log level: 4(EXCEPTION), 3(ERROR), 2(WARNING), 1(INFO), 0(DEBUG)
> os.environ['GLOG_v'] = '3'
> ```
> 
> This approach provides cleaner output while preserving important information in your notebook.

**Comment by shun-cheng** [comment]
Thank you for your suggestion！I have modified the document according to your advice. The log output level is controlled in the environment configuration section.
> Thank you very much for your submission! Here's a suggestion: Instead of setting notebook cell output to empty, you can control [MindSpore's log](https://www.mindspore.cn/docs/zh-CN/r2.6.0rc1/api_python/env_var_list.html?highlight=glog_v#%E6%97%A5%E5%BF%97) verbosity by configuring the log level through environment variables. For optimal results, place this code after installing dependencies but before importing MindSpore:
> 
> ```python
> import os
> # Set log level: 4(EXCEPTION), 3(ERROR), 2(WARNING), 1(INFO), 0(DEBUG)
> os.environ['GLOG_v'] = '3'
> ```
> 
> This approach provides cleaner output while preserving important information in your notebook.

**Comment by shun-cheng** [comment]
Thank you for your suggestion！I have modified the document according to your advice. The log output level is controlled in the environment configuration section.




𝑺𝑪𝑪
***@***.***

File: dev/ascend/snt9/2_5/基于MindSpore实现ViT病理图像识别.ipynb [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+    "import os\n",
+    "from medmnist import INFO, PathMNIST\n",
+    "import mindspore.dataset.vision as vision\n",
+    "from mindspore.dataset import GeneratorDataset\n",
+    "\n",
+    "\n",
+    "# 指定数据集\n",
+    "data_flag = 'pathmnist'\n",
+    "download_root = './data'\n",
+    "os.makedirs(download_root, exist_ok=True) \n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于MindSpore实现ViT病理图像识别.ipynb:147]: Please remove extra spaces
```suggestion
    "os.makedirs(download_root, exist_ok=True)\n",
```

File: dev/ascend/snt9/2_5/基于MindSpore实现ViT病理图像识别.ipynb [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+    "info = INFO[data_flag]\n",
+    "\n",
+    "# 直接根据标签列表计算类别数\n",
+    "labels = info['label']\n",
+    "num_classes = len(labels)\n",
+    "n_channels = info['n_channels']\n",
+    "\n",
+    "# 下载并加载数据\n",
+    "train_dataset = PathMNIST(root=download_root, split='train', download=True)\n",
+    "val_dataset = PathMNIST(root=download_root, split='val',   download=True)"
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于MindSpore实现ViT病理图像识别.ipynb:158]:
```suggestion
    "val_dataset = PathMNIST(root=download_root, split='val', download=True)"
```

File: dev/ascend/snt9/2_5/基于MindSpore实现ViT病理图像识别.ipynb [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+    "        vision.ToTensor(),\n",
+    "    ]\n",
+    "    ds_gen = ds.GeneratorDataset(dataset, column_names=['image','label'], shuffle=shuffle)\n",
+    "    ds_gen = ds_gen.map(input_columns=['image'], operations=transform)\n",
+    "    ds_gen = ds_gen.map(input_columns=['label'], operations=lambda x: x.squeeze().astype('int32'))\n",
+    "    ds_gen = ds_gen.batch(batch_size)\n",
+    "    return ds_gen\n",
+    "\n",
+    "train_ds = create_dataset(train_dataset, batch_size=64, shuffle=True)\n",
+    "val_ds   = create_dataset(val_dataset,   batch_size=64, shuffle=False)"
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/基于MindSpore实现ViT病理图像识别.ipynb:185]:
```suggestion
    "val_ds = create_dataset(val_dataset, batch_size=64, shuffle=False)"
```

**Comment by hongziqi** [comment]
Please rename file to "dev/ascend/snt9/2_5/基于ViT实现病理图像识别.ipynb"

**Review by hongziqi — APPROVED** [review]

**Comment by shun-cheng** [comment]
The file has been renamed &nbsp;to "dev/ascend/snt9/2_5/基于ViT实现病理图像识别.ipynb"




𝑺𝑪𝑪
***@***.***
